authtok.c

```diff
@@ -0,0 +1,259 @@
+#include <fido.h>
+#include <fido/types.h>
+#include <openssl/evp.h>
+#include <stdio.h>
+
+#include "authtok.h"
+#include "b64.h"
+
+static int
+encrypt_authtok(const unsigned char *plaintext, size_t plaintext_len,
+                const unsigned char *key, /* 32 bytes */
+                unsigned char *tag,       /* 16 bytes */
+                unsigned char *ciphertext /* equal to plaintext_len */
+) {
+  EVP_CIPHER_CTX *ctx = NULL;
+  EVP_CIPHER *cipher = NULL;
+  int len;
+  int retval = 0;
+  unsigned char iv[12] = {0};
```

Comment:
The nonce for AES-256-GCM doesn't have to be unpredictable. It just need to be unique for each encryption with the same key. Since each encryption key (the hmac secret) is only used once to encrypt a single auth token, there should be no risk of nonce reuse.